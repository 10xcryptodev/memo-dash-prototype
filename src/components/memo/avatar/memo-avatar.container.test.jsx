import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import MemoAvatarContainer from './memo-avatar.container'

import testUsers from '../../../test-utils/test-users'

describe('<MemoAvatarContainer />', () => {
  let store
  const alice = testUsers['alice']
  const bob = testUsers['bob']

  beforeEach(() => {
    // Mock store
    const mockStore = configureStore()

    store = mockStore({
      user: { currentUser: alice.username, users: {} }
    })

    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('Shallow rendering', () => {
    it('renders without crashing', () => {
      const wrapper = shallow(<MemoAvatarContainer memo={alice.memos[0]} />, { context: { store } })
      expect(wrapper).toMatchSnapshot()
    })

    it('foreign memo', () => {
      const wrapper = shallow(<MemoAvatarContainer memo={bob.memos[0]} />, { context: { store } })
      expect(wrapper).toMatchSnapshot()
    })
  })
})
