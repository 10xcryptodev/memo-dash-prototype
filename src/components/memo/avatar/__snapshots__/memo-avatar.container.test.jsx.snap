// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoAvatarContainer /> Shallow rendering foreign memo 1`] = `
<MemoAvatarComponent
  isMemoOfCurrentUser={false}
  memo={
    Object {
      "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg",
      "idx": 1,
      "memoDatetime": "2018-07-26T16:45:08.218Z",
      "memoLikesCount": 0,
      "memoRepliesCount": 0,
      "memoText": "Qui magnam maiores laboriosam animi sit aut.",
      "memoTipTotal": 0,
      "username": "bob",
    }
  }
  onGoToProfileClicked={[Function]}
/>
`;

exports[`<MemoAvatarContainer /> Shallow rendering renders without crashing 1`] = `
<MemoAvatarComponent
  isMemoOfCurrentUser={true}
  memo={
    Object {
      "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
      "idx": 1,
      "memoDatetime": "2018-07-26T16:44:56.000Z",
      "memoLikesCount": 1,
      "memoRepliesCount": 0,
      "memoText": "Suscipit vel nobis.",
      "memoTipTotal": 0,
      "username": "alice",
    }
  }
  onGoToProfileClicked={[Function]}
/>
`;
